writeaheadlog_integration_test.go

```diff
@@ -1,556 +1,487 @@
 	return err
 }
 
-// changeSplotch will change the targeted splotch file on disk to count down
-// form the provided value.
-func (ca *countdownArray) changeSplotch(splotchIndex, newVal uint64) error {
-	// The splotch update is intentionally encoded as the full write to
-	// simulate what a full write would be from a real application.
-	newSplotch := make([]byte, 10*8*(splotchIndex+1))
-	if newVal != 0 {
-		for i := uint64(0); i < (splotchIndex+1)*10; i++ {
-			binary.LittleEndian.PutUint64(newSplotch[8*i:(8*i)+8], newVal-i)
-		}
-	}
-	update := Update{
-		Name:    changeSplotchName,
-		Version: "1.0.0",
-		Instructions: changeSplotchUpdate{
-			index:       uint64(splotchIndex),
-			splotchData: newSplotch,
-		}.marshal(),
-	}
+// threadedSetupWrite simulates a setup by updating the checksum and writing
+// random data to a file which uses the checksum as a filename
+func (s *silo) threadedSetupWrite(done chan error, dataPath string, ncs checksum) {
+	// signal completion
+	defer close(done)
 
-	// Create the WAL transaction.
-	tx, err := ca.wal.NewTransaction([]Update{update})
+	// write new data file
+	newFile, err := s.deps.create(filepath.Join(dataPath, hex.EncodeToString(ncs[:])))
 	if err != nil {
-		return err
+		done <- err
+		return
 	}
-	// Signal completed setup and then wait for the commitment to finish.
-	// (there was no setup, but we still have to signal it)
-	errChan := tx.SignalSetupComplete()
-	err = <-errChan
+	_, err = newFile.Write(fastrand.Bytes(10 * pageSize))
 	if err != nil {
-		return err
+		done <- err
+		return
 	}
-	// Apply the updates.
-	err = ca.applyUpdates([]Update{update})
-	if err != nil {
-		return err
-	}
-	err = tx.SignalUpdatesApplied()
-	if err != nil {
-		return err
+	syncErr := newFile.Sync()
+	if err := newFile.Close(); err != nil {
+		done <- err
+		return
 	}
-	return nil
-}
 
-// changeSplotchApply applies a change splotch update to the countdown array.
-func (ca *countdownArray) changeSplotchApply(update Update) error {
-	csu := unmarshalChangeSplotchUpdate(update.Instructions)
-	offset := int64(csu.index*(csu.index+1)/2) * 8 * 10
-	_, err := ca.splotchFile.WriteAt(csu.splotchData, offset)
-	return err
+	// sync changes
+	done <- build.ComposeErrors(syncErr)
 }
 
-// applyUpdates will apply a bunch of wal updates to the ca persist file.
-func (ca *countdownArray) applyUpdates(updates []Update) error {
-	for _, update := range updates {
-		if update.Name == addCountName {
-			err := ca.addCountApply(update)
-			if err != nil {
-				return nil
+// threadedUpdate updates a silo 1000 times and leaves the last transaction unapplied.
+func (s *silo) threadedUpdate(t *testing.T, w *WAL, dataPath string, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	// Allocate some memory for the updates
+	sus := make([]*siloUpdate, 0, len(s.numbers))
+	for {
+		// Change between 1 and len(s.numbers)
+		length := fastrand.Intn(len(s.numbers)) + 1
+		var ocs = s.checksum()
+		var ncs checksum
+		var appendFrom = length
```

Comment:
more typical would be `appendFrom := length`

Reply:
Yes, but I can't stand a block of variable declarations which use both styles. :P